src/main/java/uk/gov/ons/census/fwmt/csvservice/implementation/addresscheck/AddressCheckConverterService.java

```diff
@@ -77,19 +89,48 @@ public void convertToCanonical() throws GatewayException {
   }
 
   private void processObject(CsvToBean<AddressCheckListing> csvToBean) throws GatewayException {
-    // Gonna assume no one is gonna be stupid and run this without loading lookup
     for (AddressCheckListing addressCheckListing : csvToBean) {
-      if (postcodeLookupMap.containsKey(addressCheckListing.getPostcode().replaceAll("\\s+","").toUpperCase())) {
-        CreateFieldWorkerJobRequest createFieldWorkerJobRequest = createAddressCheckJob(addressCheckListing,
-            postcodeLookupMap.get(addressCheckListing.getPostcode().replaceAll("\\s+","").toUpperCase()));
-        gatewayActionAdapter.sendJobRequest(createFieldWorkerJobRequest, CANONICAL_ADDRESS_CHECK_CREATE_SENT);
-        gatewayEventManager.triggerEvent(String.valueOf(createFieldWorkerJobRequest.getCaseId()),
-            CSV_ADDRESS_CHECK_REQUEST_EXTRACTED);
-      } else {
+      if (postcodeLookupMap.containsKey(getPostcode(addressCheckListing))) {
+        if (addressCheckListingIsValid(postcodeLookupMap.get(getPostcode(addressCheckListing)))) {
+          CreateFieldWorkerJobRequest createFieldWorkerJobRequest = createAddressCheckJob(addressCheckListing,
```

Comment:
It'll be easier to read if you put the chunk between this if and the event trigger in its own function i.e.createAndSendJob(AddressCheckListing addressCheckListing, String postcode)